src/manage_nsfs/nc_lifecycle.js

```diff
@@ -490,26 +501,50 @@ class NCLifecycle {
         return [];
     }
 
+    /**
+     * loads the objects list batch using list_objects and updates the rule state
+     * @param {Object} object_sdk
+     * @param {Object} lifecycle_rule
+     * @param {Object} bucket_json
+     * @param {Object} rule_state
+     * @returns
+     */
+    async load_objects_list(object_sdk, lifecycle_rule, bucket_json, rule_state) {
+        const objects_list = await object_sdk.list_objects({
+            bucket: bucket_json.name,
+            prefix: lifecycle_rule.filter?.prefix,
+            key_marker: rule_state.key_marker,
+            limit: config.NC_LIFECYCLE_LIST_BATCH_SIZE
+        });
+        if (objects_list.is_truncated) {
+            rule_state.key_marker = objects_list.next_marker;
```

Comment:
shouldn't this be under rule_state.expire.key_marker? where do we set noncurrent/expire on the rule state?

Reply:
those are noncurrent/expire. the I changed the variable names